From 0e8d8ed494433237cff0ad6c513c40f025005e8c Mon Sep 17 00:00:00 2001
From: Jun Zhang <jun.zhang@windriver.com>
Date: Fri, 16 Oct 2015 10:48:21 +0800
Subject: [PATCH] Fix symlink-bad-length test for 64-bit architectures.

* src/util.c: Return non-zero exit code if EOF is hit prematurely.

Backport commit:
http://git.savannah.gnu.org/cgit/cpio.git/commit/src/util.c?id=f6a8a2cbd2d5ca40ea94900b55b845dd5ca87328

Upstream-Status: Backport

Signed-off-by: Jun Zhang <jun.zhang@windriver.com>
---
 src/util.c | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/src/util.c b/src/util.c
index 00953d5..b536d82 100644
--- a/src/util.c
+++ b/src/util.c
@@ -206,10 +206,7 @@ tape_fill_input_buffer (int in_des, int num_bytes)
   if (input_size < 0)
     error (1, errno, _("read error"));
   if (input_size == 0)
-    {
-      error (0, 0, _("premature end of file"));
-      exit (1);
-    }
+    error (PAXEXIT_FAILURE, 0, _("premature end of file"));
   input_bytes += input_size;
 }
 
-- 
1.9.1

